cpp/src/io/json/json_tree.cu

```diff
@@ -201,45 +237,63 @@ tree_meta_t get_tree_representation(device_span<PdaTokenT const> tokens,
                       return is_node(tokens_gpu[i]);
                     });
   CUDF_EXPECTS(node_range_out_end - node_range_out_it == num_nodes, "node range count mismatch");
+  nvtxRangePop();
 
+  nvtxRangePushA("node_parent_ids");
   // Node parent ids: previous push token_id transform, stable sort, segmented scan with Max,
   // reorder, copy_if. This one is sort of logical stack. But more generalized.
   // TODO: make it own function.
   rmm::device_uvector<size_type> parent_token_ids(num_tokens, stream);
   rmm::device_uvector<size_type> initial_order(num_tokens, stream);
+  nvtxRangePushA("seq, tabulate");
   thrust::sequence(rmm::exec_policy(stream), initial_order.begin(), initial_order.end());
   thrust::tabulate(rmm::exec_policy(stream),
                    parent_token_ids.begin(),
                    parent_token_ids.end(),
                    [does_push, tokens_gpu = tokens.begin()] __device__(auto i) -> size_type {
                      return (i > 0) && does_push(tokens_gpu[i - 1]) ? i - 1 : -1;
+                     // XXX: How is this algorithm working for JSON lines?
```

Comment:
https://stackoverflow.com/questions/1452934/what-is-the-meaning-of-xxx-in-code-comments
It's meant to be hints for future development. 
This line can be removed because JSON-lines feature is implemented already.